url/url.cpp

```diff
@@ -1460,14 +1464,16 @@ std::optional<std::tuple<std::uint64_t, bool>> UrlParser::parse_ipv4_number(std:
         }
     }
 
+    // TODO(zero-one): Differ width based on largest integer value supported by platform?
     std::uint64_t out;
 
     auto res = std::from_chars(input.data(), input.data() + input.size(), out, r);
 
+    // This deviation from the spec is necessary, because the spec assumes arbitrary precision
     if (res.ec == std::errc::invalid_argument || res.ec == std::errc::result_out_of_range) {
-        spdlog::info("Invalid ipv4 number");
+        spdlog::info("IPv4 number > 2^64");
 
-        return std::nullopt;
+        return {{out, true}};
```

Comment:
`out` isn't modified on error, so this is returning garbage: https://en.cppreference.com/w/cpp/utility/from_chars#Return_value

I can't imagine the tests would pass if this is ever read. Can you return something something saner?